app/models/assignment.rb

```diff
@@ -338,7 +337,7 @@ def update_remark_request_count
 
   def total_criteria_weight
     factor = 10.0 ** 2
-    (get_criteria.sum('weight') * factor).floor / factor
+    (get_criteria.map(&:weight).sum() * factor).floor / factor
```

Comment:
The sum method (when given an argument) can be applied to add up values in a particular column in the database. 'weight' used to be a column, but we replaced it with 'max mark'.